Sprint 17

Hajime4life opened the pull request.

Hajime4life committed:
Добавления:
- Добавлен Pod файл
- Интегрирована яндекс метрика

Hajime4life committed:
Добавления:
- Добавлен функционал поисковой строки для фильтрации треков
- Добавлнена кнопка фильтра треков (+функционал)
- Добавлен экран статистики и функционал на нем
- Внедрен сервис аналитики от яндекс
- Реализована локализация на 3 языках

Hajime4life committed:
Добавления:
- Доработано контекстное меню

Hajime4life committed:
Cохраняюсь перед тестами

Hajime4life committed:
Добавление:
- Добавлены скриншот тесты

Hajime4life committed:
Добавления:
- Добавлен .gitigonre чтобы тесты не спамили в репозиторий скриншоты
- Добавлена темная тема

NaR0RoW commented on Tracker.xcworkspace/xcuserdata/mac.xcuserdatad/IDEFindNavigatorScopes.plist:
Папку `xcuserdata` стоит игнорировать, кладя в `.gitignore`. В ней хранится информация о состоянии проекта (какой файл открыт, какой таргет выбран и т.п.), брейкпоинты и схемы для текущего пользователя `Xcode`

Hajime4life replied:
xcuserdata у меня в гитигноре, но проблема в том что гит игнор я поздно завел

NaR0RoW commented on Tracker/App/AppDelegate.swift:
```diff
@@ -1,10 +1,16 @@
+import YandexMobileMetrica
 
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
```
Стоит добавить модификатор `final`, чтобы вызовы методов работали быстрее.
Но это уже со звездочкой, хотя на собеседованиях спрашивают. Можно почитать про [диспачеризацию тут](https://habr.com/ru/articles/673400/)

NaR0RoW commented on Tracker/Controllers/Alert/AlertPresenterProtocol.swift:
```diff
@@ -0,0 +1,5 @@
+import UIKit
```
Лучше не импортировать `UIKit` в те файлы, где нет работы с UI, а воспользоваться импортом `Foundation`

NaR0RoW commented on Tracker/Controllers/DefaultController.swift:
```diff
@@ -2,19 +2,47 @@ import UIKit
+        case newEvents = "screen.newEvents"
+        
+        var text: String {
+            NSLocalizedString(self.rawValue, comment: "")
```
`self` везде лучше опускать, если явно того не требуется

NaR0RoW commented on Tracker/Controllers/DefaultController.swift:
```diff
@@ -37,33 +65,79 @@ class DefaultController: UIViewController {
         
+        var symbol: String {
+            switch self {
+                case .emojiSmile: return "🙂"
```
В однострочных выражениях `return` можно опускать

NaR0RoW commented on Tracker/Controllers/DefaultController.swift:
```diff
@@ -37,33 +65,79 @@ class DefaultController: UIViewController {
     
     enum ImageNames: String {
         case dizzy = "dizzy"
+        case placeholderStatistik = "placeholderStatistik"
```
Ошибка в правописании:
`Statistik` -> `Statistic`

Лучше включить [настройку](https://sarunw.com/posts/spell-checking-in-xcode/), чтобы `Xcode` подсвечивал ошибки в правописании

NaR0RoW commented on TrackerTests/__Snapshots__/TrackerTests/testOnboardingFirstPage.1.png:
Генерируемые сторонними библиотеками файлы и папки лучше выносить в `.gitignore`

NaR0RoW commented on Pods/Pods.xcodeproj/project.pbxproj:
Всю папку `Pods/` нужно положить в `.gitignore`, чтобы она не лежала в репозитории. Зависимости ревьюер или другие участники команды смогут собрать, выполнив у себя команду `pod install`. Эта команда посмотрит на файлики с зависимостями `Podfile` и `Podfile.lock` и по ним все скачает и разложит для сборки приложения

Hajime4life replied:
# CocoaPods
# We recommend against adding the Pods directory to your .gitignore. However
# you should judge for yourself, the pros and cons are mentioned at:
# https://guides.cocoapods.org/using/using-cocoapods.html#should-i-check-the-pods-directory-into-source-control
Pods/
# Add this line if you want to avoid checking in source code from the Xcode workspace
*.xcworkspace

лежит, но я забыл

NaR0RoW commented:
После редактирования привычки на главном экране остается старое название:

![simulator_screenshot_151E11AB-1D78-4891-9274-53E9D23DE11C](https://github.com/user-attachments/assets/2897a934-311a-4a87-898c-ddb9c6962e31)

NaR0RoW commented:
Закрепленная привычка должна быть первой в коллекции (т.е. находиться на самом верху). Сейчас же просто добавляется значок пинна:

![simulator_screenshot_DF4A814C-A878-44B2-A714-08D35F47E0C1](https://github.com/user-attachments/assets/f2edba3a-504b-4350-a9a9-716b8fe1d52d)

NaR0RoW commented:
У кнопки "Фильтры" и у лейбла количества дней в треках на главном экране не отображается перевод:

![simulator_screenshot_29A54C29-6C62-4C21-8995-38A874A1A247](https://github.com/user-attachments/assets/12be8ebd-d4de-4f96-9d57-98527c64e22a)

Hajime4life committed:
Исправления:
- Исправлено закрепление, теперь при закреплении происходит обновление интерфейса
- Исправлена логика работы календаря
- Учтены замечания ревьюера

Hajime4life committed:
Исправлена темная тема

Hajime4life committed:
- Кнопки фильтров тепреь локализуются корректно

Hajime4life committed:
- Исправлена локализация дропдаун кнопок

Hajime4life committed:
- Учтены все недочеты ревьюера. (готов к ревью)

Hajime4life commented:
На календаре ожидается двойное наджатие на ячейку
